sdk/eventhub/event-hubs/src/partitionPump.ts

```diff
@@ -0,0 +1,89 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import * as log from "./log";
+import { EventProcessorOptions, PartitionProcessor } from "./eventProcessor";
+import { PartitionContext } from "./partitionContext";
+import { EventHubClient } from "./eventHubClient";
+import { EventPosition } from "./eventPosition";
+import { EventHubConsumer } from "./receiver";
+import { AbortController } from "@azure/abort-controller";
+
+export class PartitionPump {
+  private _partitionContext: PartitionContext;
+  private _eventHubClient: EventHubClient;
+  private _partitionProcessor: PartitionProcessor;
+  private _processorOptions: EventProcessorOptions;
+  private _receiver: EventHubConsumer | undefined;
+  private _isReceiving: boolean = false;
+  private _abortController: AbortController;
+
+  constructor(
+    eventHubClient: EventHubClient,
+    partitionContext: PartitionContext,
+    partitionProcessor: PartitionProcessor,
+    options?: EventProcessorOptions
+  ) {
+    if (!options) options = {};
+    this._eventHubClient = eventHubClient;
+    this._partitionContext = partitionContext;
+    this._partitionProcessor = partitionProcessor;
+    this._processorOptions = options;
+    this._abortController = new AbortController();
+  }
+
+  async start(partitionId: string): Promise<void> {
+    if (this._partitionProcessor.initialize) {
```

Comment:
Here it would be safer to replace this check with:
```suggestion
    if (typeof this._partitionProcessor.initialize === "function") {
```

Reply:
Added the same check when we create `partitionProcessor`.